docs/explanation/classic-confinement.rst

```diff
@@ -2,3 +2,108 @@
+build-time.
+
+
+Requirements
+------------
+
+Snapcraft determines confinement based on the value of the ``confinement`` key in the
+project file. For classic confinement, it must be set as ``confinement: classic``. There
+are no further requirements for the configuration -- when Snapcraft builds the snap, it
+won't be sandboxed. That covers the build and local side of the confinement.
+
+Once the snap is prepared, if the snap author wishes to distribute it through official
+channels, they must :ref:`apply for classic confinement
+<how-to-enable-classic-confinement-request-snap-store>`.
+
+
+Build-time process
+------------------
+
+Snapcraft builds classic snaps differently from snaps with strict confinement.
+
+This is because in order to execute correctly, packages in classic snaps require
+dynamic executables to load shared libraries from the appropriate base snap instead of
+using the host's root filesystem.
+
+To prevent incompatibilities, binaries in classic snaps must be built with appropriate
+linker parameters, or patched to allow loading shared libraries from their base snap. In
+case of potential dynamic linking issues, the snap author must be aware that their
+package may not run as expected.
+
+There are multiple ways dynamic linking parameters can be manipulated:
+
+.. image:: https://assets.ubuntu.com/v1/24ce3093-confinement_03.png
+    :alt: The flow diagram for dynamic linking parameteres in the snap confinement process at build-time.
+
+- **Runtime library paths**. The dynamic section of an ELF file contains the RPATH
+  entry, which lists the runtime paths to shared libraries to be searched before the
+  paths set in the LD_LIBRARY_PATH environment variable. Multiple paths separated by a
+  colon can be specified.
+- ``$ORIGIN`` **path**. The special value ``$ORIGIN`` represents the path where the
+  binary is located, thus allowing the runtime library path to be set relative to that
+  location. For example, ``$ORIGIN/../lib`` for an executable installed under ``bin/``
+  with libraries in ``lib/``.
+- **File interpreter**. The special ELF section ``.interp`` holds the path to the
+  program interpreter. If used, it must be set to the path of the appropriate dynamic
+  linker -- the dynamic linker from the snap package being created If libc is staged, or
+  the dynamic linker provided by the base snap otherwise. Usually, the program
+  interpreter is provided by the base, but it can also be provided by the snap. This
+  happens before any library resolution takes place.
+
+To execute as expected, binaries in a classic snap app must be configured to look for
+shared libraries provided by the base snap or bundled as part of the app snap. This is
+achieved by setting the runtime path to shared libraries in all ELF binaries (except
+relocatable object files) that are present in the package payload.
+
+- The ``$RPATH`` value must be set to reach all needed entries in the dynamic section of
+  the ELF binary.
```

Comment:
It took me a minute to understand what this list was describing.  It might be worth explaining this list is what Snapcraft does for classic snaps (specifically with `build-attributes: [enable-patchelf]` defined in a part):

```suggestion
relocatable object files) that are present in the package payload. Snapcraft uses patchelf to ensure:

- The ``$RPATH`` value is set to reach all needed entries in the dynamic section of
  the ELF binary.
```

Reply:
This was disaggregated from one huge document, so it makes sense that I'd make a mistake like this. I expanded on your solution: c51b5ee